ChainRouter.php

```diff
@@ -306,4 +306,25 @@ public function getRouteCollection()
 
         return $this->routeCollection;
     }
+    
+    /**
+     * Identify if any routers have been added
+     * 
+     * @return bool
+     */
+    public function hasRouters()
+    {
+        return !empty($this->routers);
+    }
+    
+    /**
+     * Get the number of added routers
+     * 
+     * @return integer
+     */
+    public function getNumRouters()
+    {
+        return count($this->routers);
```

Comment:
for this you are adding hasRouters. as said, the `all` caches the result, so unless you do have routers in the chainrouter but do not execute the chainrouter, there is about no additional cost in calling `all` repeatedly.